### ISSUE TITLE
Socket io server v4 and swift client 16.0.1

### ISSUE THREAD

**Issue by JimmyBjorklund** [socketio/socket.io-client-swift#1355]
Hi We have a socket io server running v4 but the swift client cannot connect. Is this expected? 
The server accepts connections from android and node justfine. 
We have allowEIO3: false on the server.
The swift client get the upgrade message but after that nothing happens. 

Thanks for any help.
/Jimmy

**Comment by vandadnp** [comment]
Same thing here, this is quite a dealbreaker for us

**Comment by nuclearace** [comment]
Yes I’m looking into this. A new version should be coming soon

**Comment by vandadnp** [comment]
@nuclearace hi. Do you have any updates on this please?

**Comment by vandadnp** [comment]
@nuclearace hi there again. Do you have any updates on this please?

**Comment by vandadnp** [comment]
@nuclearace any updates on this would be really helpful. thank you for your efforts and time

**Comment by vandadnp** [comment]
@nuclearace hi again, since 20 days have gone by, I thought to ask you for some pointers as to where the bug lies and we could try to solve it ourselves and send a PR to you if that would help?

**Comment by neobie** [comment]
There are so many versions of socket IO which are not compatible with each other ?
I am trying to connect IOS to socket v4 server as well.

**Comment by Sadmansamee** [comment]
We really stuck on this :(

**Comment by vandadnp** [comment]
@Sadmansamee agreed, we too. The state of this repository is quite alarming. We are ready to help fix this ourselves but absolutely zero communication so far from @nuclearace unfortunately

**Comment by darrachequesne** [comment]
Hi everyone! I was not able to reproduce the issue: https://github.com/socketio/socket.io-swift-fiddle

Am I missing something? Could it be a version mismatch?

**Comment by JimmyBjorklund** [comment]
No we use the lates on the backen and the latest of this client should work without the allowEIO3, this work great for node clients and android it only the swift client that is not working

**Comment by JimmyBjorklund** [comment]
> Hi everyone! I was not able to reproduce the issue: https://github.com/socketio/socket.io-swift-fiddle
> 
> Am I missing something? Could it be a version mismatch?

You ar missing to set the options:
    this.io = new SocketIO.Server(httpServer, {
      path: path,
      allowEIO3: false
    });

**Comment by vandadnp** [comment]
@darrachequesne have you seen the above message from @JimmyBjorklund ?

**Comment by Pic2490** [comment]
Should we just downgrade versions? I'm having the same problem as well

**Comment by darrachequesne** [comment]
@JimmyBjorklund hmm, that's weird, on the server side:

- `allowEIO3` defaults to `false` ([doc](https://socket.io/docs/v4/server-initialization/#allowEIO3))
- `path` defaults to `"/socket.io/"`, which is the same value as the client [here](https://github.com/socketio/socket.io-client-swift/blob/8d9a6138356a933a433c4454d28e01747f9f2935/Source/SocketIO/Manager/SocketManager.swift#L573) ([doc](https://socket.io/docs/v4/server-initialization/#path))

That being said, the `path` must match on the server and the client. Are you using a custom path?

@Pic2490 please check the compatibility table here: https://github.com/socketio/socket.io-client-swift/blob/master/Usage%20Docs/Compatibility.md

**Comment by rouzbeh-abadi** [comment]
The swift client issue is still not fixed, and adding `.connectParams(["EIO": "3"])` doesn't help that. Should we wait for an update?

**Comment by Ryce** [comment]
Wow, I spent the last 2 days pulling my hair on this one. And downgrading *both* iOS and Node to EIO3 solved the issue...
